lambdas/functions/management/management.go

```diff
@@ -0,0 +1,59 @@
+package management
+
+import (
+	"context"
+	"fmt"
+	"github.com/weka/aws-tf/modules/deploy_weka/lambdas/functions/weka_api"
+
+	"github.com/weka/go-cloud-lib/connectors"
+	"github.com/weka/go-cloud-lib/lib/jrpc"
+	"github.com/weka/go-cloud-lib/lib/weka"
+	"github.com/weka/go-cloud-lib/logging"
+	"github.com/weka/go-cloud-lib/protocol"
+)
+
+func GetWekaJRPC(ctx context.Context, request weka_api.WekaApiRequest) (interface{}, error) {
+	logger := logging.LoggerFromCtx(ctx)
+	logger.Debug().Msg("GetWekaJRPC > Start")
+
+	var method weka.JrpcMethod
+	var jrpcProtocol interface{}
+	switch request.Method {
+	case "status":
+		method = weka.JrpcStatus
+		jrpcProtocol = &protocol.WekaStatus{}
+	default:
+		return nil, fmt.Errorf("management - unsupported request type: %s", request.Method)
+	}
+
+	logger.Debug().Msgf("GetWekaJRPC > Username: %s", request.Username)
+	jrpcBuilder := func(ip string) *jrpc.BaseClient {
+		return connectors.NewJrpcClient(ctx, ip, weka.ManagementJrpcPort, request.Username, request.Password)
+	}
+
+	ips := request.BackendPrivateIps
+	if len(ips) == 0 {
+		return nil, fmt.Errorf("GetWekaJRPC - backend private ips are empty")
+	}
+	logger.Debug().Msgf("GetWekaJRPC > BackendPrivateIps: %v", ips)
+
+	var params interface{}
+	if request.Params != nil {
+		params = request.Params
+	} else {
+		params = struct{}{}
+	}
+
+	jpool := &jrpc.Pool{
+		Ips:     ips,
+		Clients: map[string]*jrpc.BaseClient{},
+		Active:  "",
+		Builder: jrpcBuilder,
+		Ctx:     ctx,
+	}
+
+	if err := jpool.Call(method, params, &jrpcProtocol); err != nil {
```

Comment:
this isn't `jrpcProtocol`, but `jrpcCallResponse`

Reply:
4d60cca8682b53101f94f0e629716b150453599e